content/post/house-prices-in-king-county.Rmd

```diff
@@ -411,3 +410,26 @@ We have seen different ways to improve models with respect to our criteria by:
 
 A combination of all of the above would most likely yield an even better model.
 This is left as an exercise to the reader.
+
+The best model we found in this example is the `ranger` model with the added `med_price` feature.
+In a final step, we now want to assess the model's quality on the held-out data we stored in our
+`task_test`. In order to do so, and to prevent data leakage, we can only add the median price from 
+the training data.
+
+```{r}
+data = task_train$data(cols = c("price", "zipcode"))
+data[, med_price := median(price), by = "zipcode"]
+test_data = task_test$data(cols = "zipcode")
+test = merge(test_data, unique(data[, .(zipcode, med_price)]), all.x = TRUE)
+task_test$cbind(test)
+```
+
+Now we can use the augmented `task_test` to predict on new data.
+
+```{r}
+lrn$train(task_train)
```

Comment:
This is not the ranger learner you are talking about, right?

Reply:
you are right, fixed